### ISSUE TITLE
Fix for invalid escape sequences wrongly escapes some backslashes

### ISSUE THREAD

**Issue by Jean-Abou-Samra** [hhatto/autopep8#558]
In strings containing both an invalid escape sequence and an escaped delimiter, autopep8's aggressive mode adds a leading `r`, leading to an extraneous backslash.

```python
'\m\''
```
is converted by

```shell
$ autopep8 -a
```
into
```python
r'\m\''
```
Yet they're not equivalent:
```python
>>> print('\m\'')
\m'
>>> print(r'\m\'')
\m\'
```
In such cases, the fix should be to insert backslashes inside the string: `'\\m\''`, or change the delimiters: `r"\m'"`.
## Your Environment
* Python version: 3.8.2
* autopep8 version: autopep8 1.5.3 (pycodestyle: 2.6.0)
* Platform: Ubuntu

**Comment by Jean-Abou-Samra** [comment]
To give a simpler example, `"\m\n"` is converted into `r"\m\n"`.

**Comment by hhatto** [comment]
Thanks for feedback.

We have released version 1.5.5.
This version should fix the problem.

**Comment by Jean-Abou-Samra** [comment]
Thank you very much!
